backend/src/main/java/planit/people/preparation/Services/Service_Calendar.java

```diff
@@ -63,4 +72,151 @@ private Set<String> getAttendeeRefreshTokens(List<Entity_GoogleAccount> googleAc
+        idao_guest.saveAll(presetGuests);
+    }
+
+    /**
+     * Get Preset details for a PlanIt User. The retrieval of details will occur in the following steps:
+     * 1- get all EventPreset for the given PlanIt User. In case no result found. an error will be thrown
+     * 2- Query guests and preset availabilities from the retrieved list of EventPresets.
+     * 3- Convert all returned results from the DB into a list of PresetDetail records.
+     *
+     * @param planItUserId the PlanIt user id for whom the system is retrieving the preset details.
+     * @return a list of preset details.
+     */
+    public List<DTO_PresetDetail> getEventPresetsByPlanItUserId(Long planItUserId) {
+        List<DTO_PresetDetail> newPresets = new ArrayList<>();
+        List<Entity_EventPreset> eventPresets = idaoUser.getAllEventPresetByPlanItId(planItUserId);
+        if (eventPresets.isEmpty()) {
+            throw new EmptyResultDataAccessException(0);
+        } else {
+            Map<Entity_EventPreset, List<Entity_PresetAvailability>> availabilities = getAvailabilities(idaoPresetAvailability.getAllByIdEventPreset(eventPresets));
+            Map<Entity_EventPreset, List<Entity_Guest>> guests = getGuests(idao_guest.getAllByEntityEventPreset(eventPresets));
+            for (Entity_EventPreset eventPreset : eventPresets) {
+                newPresets.add(
+                        new DTO_PresetDetail(
+                                eventPreset,
+                                guests.get(eventPreset),
+                                availabilities.get(eventPreset))
+                );
+            }
+        }
+        return newPresets;
+    }
+
+    /**
+     * Map each guest to its own EventPreset.
+     *
+     * @param guests all guests stored in the DB for the given PlanIt user Id.
+     * @return a map of list of guests mapped to their parent preset.
+     */
+    private Map<Entity_EventPreset, List<Entity_Guest>> getGuests(List<Entity_Guest> guests) {
+        Map<Entity_EventPreset, List<Entity_Guest>> result = new HashMap<>();
+        for (Entity_Guest guest : guests) {
+            if (result.containsKey(guest.getEntity_EventPreset())) {
+                result.get(guest.getEntity_EventPreset()).add(
+                        new Entity_Guest(
+                                guest.getId_event_guest(),
+                                guest.getEmail(),
+                                guest.getObligatory()
+                        )
+                );
+            } else {
+                result.put(guest.getEntity_EventPreset(), new ArrayList<>() {
+                    {
+                        add(new Entity_Guest(
+                                guest.getId_event_guest(),
+                                guest.getEmail(),
+                                guest.getObligatory()
+                        ));
+                    }
+                });
+            }
+        }
+        return result;
```

Comment:
A lot of things here:

1. Instead of calling `guest.getEntity_EventPreset()` 3 times, just cache it once at the beginning of the loop.
2. Instead of the `if(result.containsKey())` and depending on that instantiating or appending the list, I think you could
```
var list = results.getOrDefault(cachedKey, new ArrayList<>());
list.add(new Entity_Guest(
    guest.getId_event_guest(),
    guest.getEmail(),
    guest.getObligatory()
    )
);
results.put(cachedKey, list);
```
This would make the code much clearer and shorter